### PR TITLE
Spelling and whitespace corrections

### PR DESCRIPTION
**Pull request by fulldecent**
This change fixes several whitespace and grammar issues. In scope for this PR is:

* Spelling corrections
* Grammar corrections
* Whitespaces fixes (remove trailing spaces)
* One minor fix to add `#!/bin/bash` at the top of files

Thank you to Ninutsa Nadirashvili for help in preparing this change.

### CODE REVIEW

File: zokrates_core/src/flatten/mod.rs [hunk 1]
```diff
@@ -464,9 +464,9 @@ impl<'ast, T: Field> Flattener<'ast, T> {
                 }
             },
             e => {
-                // we have array[e] with e an arbitrary expression
+                // we have array[e] with e as an arbitrary expression
```
Comment (Schaeff) [inline zokrates_core/src/flatten/mod.rs]: I don't agree with this change. The intent is similar as "f(x) with x an integer"

File: zokrates_core/src/ir/from_flat.rs [hunk 2]
```diff
@@ -67,7 +67,10 @@ impl<T: Field> From<FlatProg<T>> for Prog<T> {
         // get the main function
         let main = flat_prog.main;
 
-        // get the interface of the program, ie which inputs are private and public
+        // get the signature to keep high level information in the low level representation
```
Comment (Schaeff) [inline zokrates_core/src/ir/from_flat.rs]: This change should be removed, I suspect merging develop into your branch should work.

File: zokrates_core/src/static_analysis/inline.rs [hunk 3]
```diff
@@ -24,7 +24,7 @@ use zokrates_field::field::Field;
 /// An inliner
 #[derive(Debug)]
 pub struct Inliner<'ast, T: Field> {
-    modules: TypedModules<'ast, T>, // the modules to look for functions in when inlining
+    modules: TypedModules<'ast, T>, // the modules to look for functions when inlining
```
Comment (Schaeff) [inline zokrates_core/src/static_analysis/inline.rs]:
```suggestion
    modules: TypedModules<'ast, T>, // the modules in which to look for functions when inlining
```

File: zokrates_core_test/tests/tests/split.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
...
... "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1", "1"]
```
Comment (Schaeff) [inline zokrates_core_test/tests/tests/split.json]: I'm confused what the change is here?

File: ci/install.sh [hunk 5]
```diff
@@ -10,7 +12,7 @@ main() {
         sort=gsort  # for `sort --sort-version`, from brew's coreutils.
     fi
 
-    # Builds for iOS are done on OSX, but require the specific target to be
+    # Builds for iOS are done on macOS, but require the specific target to be
```
Comment (Schaeff) [inline ci/install.sh]: Here I'd rather keep the file as is and add a reference to https://github.com/japaric/trust which it originates from

File: zokrates_book/src/concepts/control_flow.md [hunk 6]
```diff
@@ -14,7 +14,7 @@ Arguments are passed by value.
 
 ### If expressions
 
-An if expression allows you to branch your code depending on a condition.
+An "if expression" allows you to branch your code depending on a condition.
```
Comment (Schaeff) [inline zokrates_book/src/concepts/control_flow.md]: maybe if-expression is more natural?

File: zokrates_book/src/reference/cli.md [hunk 7]
```diff
@@ -13,20 +13,20 @@ You can see an overview of the available subcommands by running
...
 
 ```sh
 ./zokrates compute-witness -a 1 2 3
 ```
 
-Computes a witness for the compiled program found at `./out` and arguments to the program.
-A witness is a valid assignment of the variables, which include the results of the computation.
-Arguments to the program are passed as a space-separated list with the `-a` flag, or over `stdin`.
+Computes a witness for the compiled program found at `./out` and computes arguments of the program.
+A witness is a valid assignment of the variables that include the results of the computation.
```
Comment (Schaeff) [inline zokrates_book/src/reference/cli.md]:
```suggestion
A witness is a valid assignment of the variables, including the results of the computation.
```

File: scripts/one_liner.sh [hunk 8]
```diff
@@ -28,10 +28,10 @@ need_cmd() {
 get_bitness() {
     need_cmd head
     # Architecture detection without dependencies beyond coreutils.
-    # ELF files start out "\x7fELF", and the following byte is
+    # ELF files start out "\x7fELF", and the following bytes are
```
[the remark this answers is not in the document]
Comment (fulldecent) [reply scripts/one_liner.sh]: Fixed in 2913f3b

File: zokrates_core/src/flatten/mod.rs (hunk 1 above)
Comment (fulldecent) [reply zokrates_core/src/flatten/mod.rs]: Fixed in 3399c92

File: zokrates_core/src/ir/from_flat.rs (hunk 2 above)
Comment (fulldecent) [reply zokrates_core/src/ir/from_flat.rs]: Fixed at e1c5b6cd422ca24f728bfc58441ff6a41f387c35

File: zokrates_core/src/static_analysis/inline.rs (hunk 3 above)
Comment (fulldecent) [reply zokrates_core/src/static_analysis/inline.rs]: Fixed at 173d1f9

File: zokrates_core_test/tests/tests/split.json (hunk 4 above)
Comment (fulldecent) [reply zokrates_core_test/tests/tests/split.json]: This removes trailing whitespace, the diff viewer on GitHub is broken

File: ci/install.sh (hunk 5 above)
Comment (fulldecent) [reply ci/install.sh]: Fixed at 93ef21f

File: zokrates_book/src/concepts/control_flow.md (hunk 6 above)
Comment (fulldecent) [reply zokrates_book/src/concepts/control_flow.md]: Fixed at f32444e

File: zokrates_book/src/reference/cli.md (hunk 7 above)
Comment (fulldecent) [reply zokrates_book/src/reference/cli.md]: Fixed at be09f6d

**Comment by fulldecent** [comment]
Agreed on all counts. Individual commits for each item referenced in individual comments.

**Pull request closed without merging**